test/topology/conftest.py

```diff
@@ -182,12 +182,11 @@ def check_pre_raft(cql):
         return False
     # In Scylla, we check Raft mode by inspecting the configuration via CQL.
     experimental_features = list(cql.execute("SELECT value FROM system.config WHERE name = 'experimental_features'"))[0].value
-    return not '"raft"' in experimental_features
-
+    return 'raft' in experimental_features
 
 @pytest.fixture(scope="function")
 def fails_without_raft(request, check_pre_raft):
-    if check_pre_raft:
+    if not check_pre_raft:
```

Comment:
I'm afraid I don't understand this patch. Above you reversed the meaning of check_pre_raft, and here you reverse the result, so how does the whole patch change anything?

Actually, it's even worse. Your patch will break running on Cassandra: On Cassandra, your check_pre_raft still returns False, so `not check_pre_raft` will be true and the test will become xfail. But it shouldn't - Cassandra does not have our pre-raft bugs on single-node clusters.